src/request.ts

```diff
@@ -172,16 +171,13 @@ class DatastoreRequest {
    *   const apiResponse = data[1];
    * });
    */
-  allocateIds(key, options, callback) {
-    if (entity.isKeyComplete(key)) {
-      throw new Error('An incomplete key should be provided.');
-    }
+  allocateIds(key: entity.Key, options: AllocateIdsOptions|number): Promise<google.datastore.v1.AllocateIdsResponse>;
+  allocateIds(key: entity.Key, options: AllocateIdsOptions|number, callback: AllocateIdsCallback): void;
+  allocateIds(key: entity.Key, options: AllocateIdsOptions|number, cb?: AllocateIdsCallback): void|Promise<google.datastore.v1.AllocateIdsResponse> {
+    if (entity.isKeyComplete(key)) throw new Error('An incomplete key should be provided.');
 
-    if (is.number(options)) {
-      options = {
-        allocations: options,
-      };
-    }
+    const callback = typeof cb === 'function' ? cb : cb!;
```

Comment:
This is confusing.  Isn't `cb` always a callback, unless its null/undefined? I think we could just not have this line of code.

Reply:
Agreed! I added this line to get past the linter. It was throwing errors because we're trying to invoke a potentially undefined callback further down in the code